inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/KafkaLoadNode.java

```diff
@@ -77,13 +93,57 @@ public String genTableName() {
     @Override
     public Map<String, String> tableOptions() {
         Map<String, String> options = super.tableOptions();
-        options.put("connector", "kafka");
         options.put("topic", topic);
         options.put("properties.bootstrap.servers", bootstrapServers);
-        options.put("format", format);
         if (getSinkParallelism() != null) {
             options.put("sink.parallelism", getSinkParallelism().toString());
         }
+        if (format instanceof JsonFormat) {
+            JsonFormat jsonFormat = (JsonFormat) format;
+            options.put("connector", "upsert-kafka");
+            options.put("key.format", jsonFormat.getFormat());
+            options.put("value.format", jsonFormat.getFormat());
+            options.put("value.json.fail-on-missing-field", jsonFormat.getFailOnMissingField() + "");
```

Comment:
It is recommended to use toString instead of concatenating empty strings

Reply:
ok